src/widget/wsingletoncontainer.cpp

```diff
@@ -0,0 +1,55 @@
+#include "widget/wsingletoncontainer.h"
+
+#include <QLayout>
+
+#include "util/assert.h"
+#include "widget/wlibrary.h"
+
+// static
+WSingletonContainer::WidgetMap WSingletonContainer::m_singletons;
+
+// static
+WSingletonContainer* WSingletonContainer::getSingleton(
+        QString objectname, QWidget* pParent) {
+   WidgetMap::const_iterator widget_it = m_singletons.find(objectname);
+    if (widget_it == m_singletons.end()) {
+        qWarning() << "ERROR: Asked for an unknown singleton widget:"
+                   << objectname;
+        return NULL;
+    }
+    return new WSingletonContainer(objectname, pParent);
+}
+
+WSingletonContainer::WSingletonContainer(QString objectname, QWidget* pParent)
+        : WWidgetGroup(pParent), m_pWidget(NULL) {
+    WidgetMap::const_iterator widget_it = m_singletons.find(objectname);
+    DEBUG_ASSERT(widget_it != m_singletons.end());
+
+    m_pWidget = *widget_it;
```

Comment:
When the widget is deleted (once its parent is deleted) then other container's pointers to the widget will be junk. You can hook the QObject::destroyed signal to get a callback and void the pointer or you can use QPointer which does this for you.